pysatMissions/methods/spacecraft.py

```diff
@@ -173,18 +163,85 @@ def project_ecef_vector_onto_sc(inst, x_label, y_label, z_label,
 
     # TODO(#65): add checks for existence of ecef labels in inst
 
-    x, y, z = OMMBV.vector.project_onto_basis(
-        inst[x_label], inst[y_label], inst[z_label],
-        inst['sc_xhat_ecef_x'], inst['sc_xhat_ecef_y'], inst['sc_xhat_ecef_z'],
-        inst['sc_yhat_ecef_x'], inst['sc_yhat_ecef_y'], inst['sc_yhat_ecef_z'],
-        inst['sc_zhat_ecef_x'], inst['sc_zhat_ecef_y'], inst['sc_zhat_ecef_z'])
-    inst[new_x_label] = x
-    inst[new_y_label] = y
-    inst[new_z_label] = z
+    xx = inst['sc_xhat_ecef_x']
+    xy = inst['sc_xhat_ecef_y']
+    xz = inst['sc_xhat_ecef_z']
+
+    yx = inst['sc_yhat_ecef_x']
+    yy = inst['sc_yhat_ecef_y']
+    yz = inst['sc_yhat_ecef_z']
+
+    zx = inst['sc_zhat_ecef_x']
+    zy = inst['sc_zhat_ecef_y']
+    zz = inst['sc_zhat_ecef_z']
+
+    inst[new_x_label] = inst[x_label] * xx + inst[y_label] * xy + inst[z_label] * xz
+    inst[new_y_label] = inst[x_label] * yx + inst[y_label] * yy + inst[z_label] * yz
+    inst[new_z_label] = inst[x_label] * zx + inst[y_label] * zy + inst[z_label] * zz
 
     if meta is not None:
         inst.meta[new_x_label] = meta[0]
         inst.meta[new_y_label] = meta[1]
         inst.meta[new_z_label] = meta[2]
 
     return
+
+
+def normalize(x, y, z):
+    """Normalize a time-series of vectors.
+
+    Parameters
+    ----------
+    x : pds.Series
+        The x-component
+    y : pds.Series
+        The y-component
+    z : pds.Series
+        The z-component
+
+    Returns
+    -------
+    xhat : pds.Series
+        The normalized x-component
+    yhat : pds.Series
+        The normalized y-component
+    zhat : pds.Series
+        The normalized z-component
+
+    """
+
+    vector = np.vstack([x, y, z])
+    xhat, yhat, zhat = vector / np.linalg.norm(vector, axis=0)
+
+    return xhat, yhat, zhat
+
+
+def cross_product(x1, y1, z1, x2, y2, z2):
+    """Cross product of two vectors, v1 x v2.
+
+    Parameters
+    ----------
+    x1 : float or array-like
+        X component of vector 1
+    y1 : float or array-like
+        Y component of vector 1
+    z1 : float or array-like
+        Z component of vector 1
+    x2 : float or array-like
+        X component of vector 2
+    y2 : float or array-like
+        Y component of vector 2
+    z2 : float or array-like
+        Z component of vector 2
+
+    Returns
+    -------
+    x, y, z
+        Unit vector x,y,z components
+
+    """
+    x = y1 * z2 - y2 * z1
+    y = z1 * x2 - x1 * z2
+    z = x1 * y2 - y1 * x2
```

Comment:
```suggestion
    vec1 = np.vstack([x1, y1, z1])
    vec2 = np.vstack([x2, y2, z2])
    x, y, z = np.cross(vec1, vec2, axis=0)
```

Reply:
I like this better.  It more clearly demonstrates why you have this helper function.

Reply:
Actually, there's probably a much nicer way of doing this at a higher level in pandas. I'm going to write this up as an issue for the next one.  For the record, the routines here are mostly copied from OMMBV to remove that dependency.

Reply:
added #96.  Experimented with fixing it now, but it's more complicated than expected.

Reply:
Adding this change here.  Will improve style at a later date.